doc/doxify.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+import argparse
+from pathlib import Path
+import re
+import os
+import shutil
+
+SUBPAGE_RE = re.compile(r"\[[^\]]+\]\s*\(([^\)]+)\.md\s*\)", re.MULTILINE)
+TAG_RE = re.compile(r"(#.*)\n", re.MULTILINE)
+
+parser = argparse.ArgumentParser(description='Modfiy .md files to be doxygen compatible')
+parser.add_argument('source_dir', type=str, help='Directory containing the .md files')
+parser.add_argument('dest_dir', type=str, help='Directory the modified files are written action')
+
+args = parser.parse_args()
+
+shutil.rmtree(args.dest_dir, ignore_errors=True)
+os.makedirs(args.dest_dir)
+
+for md_path in list(Path(args.source_dir).glob('*.md')):
+    with open(md_path, 'r') as md_file:
```

Comment:
`md_path` should be a `pathlib.Path` object here, so this can be done as `md_path.read_text()` or `md_path.read_bytes()` depending on whether you want text or bytes (I'm assuming text, since you haven't opened the file with 'rb').